api-reference/api-introduction/api-keys.md

```diff
@@ -1,32 +1,97 @@
 
 ## Obtaining your API keys
 
-View your API keys in ImageKit.io dashboard under the [developers tab](https://imagekit.io/dashboard#developers). Every account is provided with one pair of a public and private key.
+[View](https://imagekit.io/dashboard/developer/api-keys) your API keys in the ImageKit.io dashboard in the **API keys** section under the **Developer options** tab. Every account has one pair of standard public and private keys by default.
+
+![The API keys section in Developer options](../../.gitbook/assets/api-keys.png)
 
 ### **Public key**
 
 This is used to identify your account in certain client-side file upload implementations. It is not meant to be secret, and you can publish this in client-side Javascript code, or an Android or iPhone app.
 
 ### **Private key**
 
-This should be kept confidential and only stored on your servers. Private key is used to [authenticate](authentication.md) your account when you make an API request.
+This should be kept confidential and only stored on your servers. A private key is used to [authenticate](authentication.md) your account when you make an API request.
 
 ## Keeping your keys safe
 
 It is strongly recommended to keep your private key safe and confidential. To help keep your API keys secure, follow these best practices:
 
-1. Do not embed API keys directly in code. API keys that are embedded in code can be accidentally exposed to the public. For example, you may forget to remove the keys from code that you share. Instead of embedding your API keys in your applications, store them in environment variables or in files outside of your application's source tree.
+1. Do not embed API keys directly in your code. API keys that are embedded in code can be accidentally exposed to the public. For example, you may forget to remove the keys from the code that you share. Instead of embedding your API keys in your applications, store them in environment variables or in files outside of your application's source tree.
 2. Do not store API keys in files inside your application's source tree. If you store API keys in files, keep the files outside your application's source tree to help ensure your keys do not end up in your source code control system. This is particularly important if you use a public source code management system such as GitHub.
 
 ## Rolling keys
 
-If an API key is compromised, you should roll that pair and start using the newly generated keys.
+If your API key is compromised, you need to revoke or "roll" the key to block any API requests that might use that key. Doing this will revoke your API key and generate a new one with the same resource access permissions as the old one.
```

Comment:
revoke is no action in our systems. So mixing revoke and roll here is confusing as roll is indeed an action in our dashboard.

Reply:
"If an API key is compromised, you should roll that pair and start using the newly generated keys. The newly generated pair has the same resource access permissions as the old one."

Reply:
Done